packages/lab/src/avatar/internal/useLoaded.ts

```diff
@@ -0,0 +1,34 @@
+import { ImgHTMLAttributes, useEffect, useState } from "react";
+
+export function useLoaded({ src }: ImgHTMLAttributes<HTMLImageElement>) {
+  const [loaded, setLoaded] = useState<false | "loaded" | "error">(false);
+
+  useEffect(() => {
+    if (!src) {
+      return undefined;
+    }
+
+    setLoaded(false);
+
+    let active = true;
+    const image = new Image();
+    image.onload = () => {
+      if (!active) {
+        return;
+      }
+      setLoaded("loaded");
+    };
+    image.onerror = () => {
+      if (!active) {
+        return;
+      }
+      image.style.display = "none";
+      setLoaded("error");
+    };
+    return () => {
```

Comment:
should probably clean up the onload and onerror events here.

I think you can do it by unsetting the properties, but might be better to use `image.addEventListener('load', ...)` so you can do `image.removeEventListener('load', ...)`